lib/CMakeLists.txt

```diff
@@ -311,134 +132,22 @@ install(TARGETS ${LIBRARY_NAME}
 install(CODE
     "file(WRITE \"${CMAKE_INSTALL_PREFIX}/lib/usd/plugInfo.json\" \"{\n    \\\"Includes\\\": [ \\\"*/resources/\\\" ]\n}\")"
 )
 
+# -----------------------------------------------------------------------------
+# subdirectories
+# -----------------------------------------------------------------------------
+add_subdirectory(base)
+add_subdirectory(listeners)
+add_subdirectory(nodes)
+add_subdirectory(utils)
+add_subdirectory(render)
 if(UFE_FOUND)
-    # UFE Python Bindings
-    set(UFE_PYTHON_LIBRARY_NAME ufe)
-
-    add_library(${UFE_PYTHON_LIBRARY_NAME}
-        SHARED
-        ufe/wrapUtils.cpp
-    )
-
-    target_compile_definitions(${UFE_PYTHON_LIBRARY_NAME}
-        PRIVATE
-            MFB_PACKAGE_NAME=${LIBRARY_NAME}
-            MFB_ALT_PACKAGE_NAME=${LIBRARY_NAME}
-            MFB_PACKAGE_MODULE=ufe
-    )
-    set_target_properties(${UFE_PYTHON_LIBRARY_NAME} 
-        PROPERTIES 
-            PREFIX ""
-    )
-    if(IS_WINDOWS)
-        set_target_properties(${UFE_PYTHON_LIBRARY_NAME}
-            PROPERTIES
-                SUFFIX ".pyd"
-        )
-    elseif(IS_MACOSX)
-        set_target_properties(${UFE_PYTHON_LIBRARY_NAME}
-            PROPERTIES
-                SUFFIX ".so"
-        )
-    endif()
-
-    target_link_libraries(${UFE_PYTHON_LIBRARY_NAME}
-        ${LIBRARY_NAME}
-    )
-
-    if(IS_MACOSX OR IS_LINUX)
-        mayaUsd_init_rpath(rpath "${PROJECT_NAME}")
-        mayaUsd_add_rpath(rpath "../..")
-        mayaUsd_install_rpath(rpath ${UFE_PYTHON_LIBRARY_NAME})
-    endif()
-
-    install(TARGETS ${UFE_PYTHON_LIBRARY_NAME}
-        LIBRARY
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/${PROJECT_NAME}
-        RUNTIME
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/${PROJECT_NAME}
-    )
-
-    if(IS_WINDOWS)
-        install(FILES $<TARGET_PDB_FILE:${UFE_PYTHON_LIBRARY_NAME}> DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/${PROJECT_NAME} OPTIONAL)
-    endif()
-
-    install(FILES ufe/__init__.py
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/${PROJECT_NAME}
-    )
-
-    if (CMAKE_UFE_V2_FEATURES_AVAILABLE)
-        # We need an empty __init__.py file in each subfolder so it is considered a python module.
-        install(FILES ae/__init__.py DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/ufe_ae)
-        install(FILES ae/__init__.py DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/ufe_ae/usd)
-        install(FILES ae/__init__.py DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/ufe_ae/usd/nodes)
-
-        # Maya Attribute Editor python template files
-        foreach(_SUBDIR base camera material mesh scope shader xform)
-            install(FILES ae/__init__.py ae/${_SUBDIR}/ae_template.py
-                DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/ufe_ae/usd/nodes/${_SUBDIR}
-            )
-        endforeach()
-    endif()
-
-endif()
-
-if(IS_MACOSX)
-    set(_macDef OSMac_)
+add_subdirectory(ufe)
 endif()
```

Comment:
Cleaned up the top level CMake. This file was getting really large and was hard to follow up what is happening there.

All the sub-directories (base, listeners, nodes, utils,render,ufe) now have their own CMakeLists.  As of CMakeV12.4, 
The target_* directive calls don't need to be part of the same CMakeLists.txt where add_executable or add_library directives are called.